### DISCUSSION TITLE
Contexto Required Survey y unas dudas

### DISCUSSION THREAD

**Discussion by ValeMP** [Cotalker/documentation#26] [2. Questions and Answers] [answered]
Hola! tenía un par de dudas de los required survey! 

1. ¿Cual es el contexto que maneja el cotlang value, de un required survey?
2. Este contexto cambia si es un required survey para iniciar un flujo en la maquina de estado o para hacer cambio de estado?

**Comment by acedward** [comment, marked as answer]
Para la pregunta (1) aquí están los contextos:
https://doc.cotalker.com/docs/documentation/automation/admin_cotlang

@rafaelwalter nos ayudas con la (2) 
@guillevalin tenemos que nombar esta funcionalidad consistentemente en todas partes.

**Reply by rafaelwalter** [reply]
@ValeMP @acedward en el punto 2 tienen contextos distintos. Porque?

RequiredSurvey para iniciar un flujo tiene que definir cual es el mapping entre campos del survey y la task. Entonces tiene su proprio bot y es obligatorio. En ese caso el contexto es:
`{ answer: COTAnswer, meta: { parentTask: string, taskGroup: string } }`

RequiredSurvey para cambio de estado no tiene su proprio bot. Pero puede ser combinado con SurveyTriggers (funcionalidad pre-existentes a RequiredSurvey) y este ultimo tienen el mismo contexto de siempre:
`{ ...task, sentAnswer: COTAnswer }`

Por qué no hicimos un bot para RequiredSurvey en cambios de estado?
Por que en cambios de estado podemos ocupar RequiredSurvey sin bots, y los cambios serán automaticos.
Asi como podemos ocupar SurveyTriggers sin RequiredSurvey.
Pero al combinar un RequiredSurvey con SurveyTrigger podemos bloquear los cambios de estado hasta que se envie el formulario y ejecutar un bot (SurveyTrigger) al enviarlo.

**Reply by guillevalin** [reply]
@acedward el nombre comercial del feature será "Required Survey" ? Efectivamente todavía tenemos inconsistencias entre "Internal Forms" y "Required Survey" a pesar de que son lo mismo.

**Reply by ValeMP** [reply]
buenisimo gracias!!!
